tests/tasks/setup_candlepin.yml

```diff
@@ -22,6 +22,31 @@
       set_fact:
         _cp_url_owner: "{{ _cp_url }}/owners/{{ lsr_rhc_test_data.reg_organization }}"  # yamllint disable-line
 
+    - name: Set helper fact with start of podman run invocation
+      set_fact:
+        podman_run_cli:
+          - podman
+          - run
+          - --rm
+          - --detach
+          - --hostname
+          - "{{ lsr_rhc_test_data.candlepin_host }}"
+          - --name
+          - candlepin
+          - --publish
+          - 8443:8443
+          - --publish
+          - 8080:8080
+          - "{{ '--privileged'
+            if (ansible_distribution in ['CentOS', 'RedHat']
+                and ansible_distribution_major_version | int < 8)
+            else '' }}"
+          - ghcr.io/ptoscano/candlepin-unofficial
+
+    - name: Filter empty items from podman run invocation
+      set_fact:
+        podman_run_cli: "{{ podman_run_cli | reject('match', '^$') | list }}"
```

Comment:
That's cool, if it works you can still pass both set_facts as `vars` in the `Start Candlepin container` to avoid setting facts.

Reply:
Why won't you do vars like this? From what I see in the PR, podman_run_cli is not used anywhere else.
```
      vars:
        podman_run_cli:
          - podman
          - run
          - --rm
          - --detach
          - --hostname
          - "{{ lsr_rhc_test_data.candlepin_host }}"
          - --name
          - candlepin
          - --publish
          - 8443:8443
          - --publish
          - 8080:8080
          - "{{ '--privileged'
            if (ansible_distribution in ['CentOS', 'RedHat']
                and ansible_distribution_major_version | int < 8)
            else '' }}"
          - ghcr.io/ptoscano/candlepin-unofficial
        podman_run_cli: "{{ podman_run_cli | reject('match', '^$') | list }}"
```

Reply:
Sorry, ansible doesn't like that we use variables with the same name. This works for me:
```
- name: debug
  vars:
    podman_run_cli_full:
      - podman
      - run
      - --rm
      - --detach
      - --hostname
      - "{{ lsr_rhc_test_data.candlepin_host }}"
      - --name
      - candlepin
      - --publish
      - 8443:8443
      - --publish
      - 8080:8080
      - "{{ '--privileged'
        if (ansible_distribution in ['CentOS', 'RedHat']
        and ansible_distribution_major_version | int < 8)
        else '' }}"
      - ghcr.io/ptoscano/candlepin-unofficial
    podman_run_cli: "{{ podman_run_cli_full | reject('match', '^$') | list }}"
  debug:
    var: podman_run_cli

```

Reply:
I tried that too, however it doesn't work either:
```
ok: [localhost] => {
    "podman_run_cli": "VARIABLE IS NOT DEFINED!"
}
```
I'm using ansible-core 2.16.9.

Reply:
How about this:
```
- name: debug
  vars:
    podman_run_cli_full:
      - podman
      - run
      - --rm
      - --detach
      - --hostname
      - candlepin_host
      - --name
      - candlepin
      - --publish
      - 8443:8443
      - --publish
      - 8080:8080
      - "{{ '--privileged'
        if (ansible_distribution in ['CentOS', 'RedHat']
        and ansible_distribution_major_version | int < 8)
        else '' }}"
      - ghcr.io/ptoscano/candlepin-unofficial
    podman_run_cli: "{{ podman_run_cli_full | reject('match', '^$') | list }}"
  debug:
    var: podman_run_cli
```

Previous didn't work because lsr_rhc_test_data is not defined. But will work in the context of the role.

Reply:
thanks for the hints! I got a working version of it, simplified a little bit